src/Meilisearch/SearchResult.cs

```diff
@@ -7,7 +7,7 @@ namespace Meilisearch
     /// Wrapper for Search Results.
     /// </summary>
     /// <typeparam name="T">Hit type.</typeparam>
-    public class SearchResult<T>
+    public class SearchResult<T> : ISearchable<T>
```

Comment:
How did you choose the name `ISearchable`? Just to know I'm really bad at naming

Reply:
https://en.wikibooks.org/wiki/C_Sharp_Programming/Interfaces#:~:text=A%20CONVENTION%20used%20in%20the,()%20in%20the%20above%20example.

> A CONVENTION used in the .NET Framework (and likewise by many C# programmers) is to place an "I" at the beginning of an interface name to distinguish it from a class name. Another common interface naming convention is used when an interface declares only one key method, such as Draw() in the above example. The interface name is then formed as an adjective by adding the "...able" suffix. So, the interface name above could also be IDrawable. This convention is used throughout the .NET Framework.